x-pack/plugins/actions/server/create_unsecured_execute_function.ts

```diff
@@ -0,0 +1,153 @@
+} from './types';
+import { ACTION_TASK_PARAMS_SAVED_OBJECT_TYPE } from './constants/saved_objects';
+import { ExecuteOptions as ActionExecutorOptions } from './lib/action_executor';
+import { extractSavedObjectReferences, isSavedObjectExecutionSource } from './lib';
+import { RelatedSavedObjects } from './lib/related_saved_objects';
+
+// This allowlist should only contain connector types that don't require API keys for
+// execution.
+const ALLOWED_CONNECTOR_TYPE_IDS = ['.email'];
+interface CreateBulkUnsecuredExecuteFunctionOptions {
+  taskManager: TaskManagerStartContract;
+  connectorTypeRegistry: ConnectorTypeRegistryContract;
+  preconfiguredConnectors: PreconfiguredConnector[];
+}
+
+export interface ExecuteOptions extends Pick<ActionExecutorOptions, 'params' | 'source'> {
+  id: string;
+  relatedSavedObjects?: RelatedSavedObjects;
+}
+
+interface ActionTaskParams extends Pick<ActionExecutorOptions, 'params'> {
+  actionId: string;
+  apiKey: string | null;
+  relatedSavedObjects?: RelatedSavedObjects;
+}
+
+export type BulkUnsecuredExecutionEnqueuer<T> = (
+  internalSavedObjectsRepository: ISavedObjectsRepository,
+  actionsToExectute: ExecuteOptions[]
+) => Promise<T>;
+
+export function createBulkUnsecuredExecutionEnqueuerFunction({
+  taskManager,
+  connectorTypeRegistry,
+  preconfiguredConnectors,
+}: CreateBulkUnsecuredExecuteFunctionOptions): BulkUnsecuredExecutionEnqueuer<void> {
+  return async function execute(
+    internalSavedObjectsRepository: ISavedObjectsRepository,
+    actionsToExecute: ExecuteOptions[]
+  ) {
+    const connectorTypeIds: Record<string, string> = {};
+    const connectorIds = [...new Set(actionsToExecute.map((action) => action.id))];
+
+    const notPreconfiguredConnectors = connectorIds.filter(
+      (connectorId) =>
+        preconfiguredConnectors.find((connector) => connector.id === connectorId) == null
+    );
+
+    if (notPreconfiguredConnectors.length > 0) {
+      throw new Error(
+        `${notPreconfiguredConnectors.join(
+          ','
+        )} are not preconfigured connectors and can't be scheduled for unsecured actions execution`
+      );
+    }
+
+    const connectors: PreconfiguredConnector[] = connectorIds
+      .map((connectorId) =>
+        preconfiguredConnectors.find((pConnector) => pConnector.id === connectorId)
+      )
+      .filter(Boolean) as PreconfiguredConnector[];
+
+    connectors.forEach((connector) => {
+      const { id, actionTypeId } = connector;
+      if (!connectorTypeRegistry.isActionExecutable(id, actionTypeId, { notifyUsage: true })) {
+        connectorTypeRegistry.ensureActionTypeEnabled(actionTypeId);
+      }
+
+      if (!ALLOWED_CONNECTOR_TYPE_IDS.includes(actionTypeId)) {
+        throw new Error(
+          `${actionTypeId} actions cannot be scheduled for unsecured actions execution`
+        );
+      }
+
+      connectorTypeIds[id] = actionTypeId;
+    });
+
+    const actions = await Promise.all(
```

Comment:
Probably I miss something but why do we need to `Promise.all` if we do not `await` inside the async function?

Reply:
You're right! I copied it from the `create_execute_function` but it looks like it's not needed there either 🙈 . I will update it in my PR: https://github.com/elastic/kibana/pull/143282